doc/i18n.md

```diff
@@ -0,0 +1,76 @@
+## Using the I18n module
+
+**This document does not apply to apps that are migrated from v1 of the Zendesk
+App Framework.**
+
+The I18n module provides a helper called `I18n.t` that you can use to localize
+your application.
+
+### Setup
+
+Put a `defaultLocale` property into your `manifest.json` file.
+
+Add your translation files as `src/translations/XX.json` where `XX` is a locale such
+as `en-US`, `de`, or `zh`.
+
+A simple translation file might look like this:
+
+```json
+{
+  "hello": "Hello!",
+  "goodbye": "Bye {{name}}!",
+  "formal": {
+    "farewell": "Farewell, friend."
+  }
+}
+```
+
+### Initialization
+
+When you know which locale you want to use, call `I18n.loadTranslations(locale)` where
+`locale` is a string like `en-US`, `de`, `zh`. This will load the strings under the
+matching file in `src/translations`. For example, you could use
+
+```javascript
+const zafClient = ZAFClient.init();
+zafClient.get('currentUser.locale').then(function(data) {
+  I18n.loadTranslations(data['currentUser.locale']);
+});
+```
+
+## Usage
+
+### I18n.loadTranslations(locale)
+
+Sets the locale to be used by `I18n.t()`
+
+#### Arguments
+
+* `locale`: String representing the filename of the required translation JSON file.
+
+#### Returns
+
+`undefined`
```

Comment:
I wouldn't include the `Returns` header if it doesn't return anything meaningful.